modules/apps/portal-workflow/portal-workflow-api/src/main/java/com/liferay/portal/workflow/portlet/tab/BaseWorkflowPortletTab.java

```diff
@@ -79,6 +89,63 @@ protected Log getLog() {
 		return _logs.get(clazz);
 	}
 
+	protected String getMVCPathAttributeName(String namespace) {
+		return StringBundler.concat(
+			namespace, StringPool.PERIOD,
+			MVCRenderConstants.MVC_PATH_REQUEST_ATTRIBUTE_NAME);
+	}
+
+	protected String getPath(
```

Comment:
@MrJohn1911 Some of the methods, like `getPath`, is only used by one of the sub-classes. Let's push such methods down.